StantDrawing/BrushTool/GroupSizeView.swift

```diff
@@ -0,0 +1,287 @@
+        iconView.layer.cornerRadius = iconView.frame.width / 2
+        
+        let originIconImage = CGPoint(x: (groupViewWidth - 25) / 2, y: 18)
+        let sizeIconImage   = CGSize(width: 25, height: 25)
+        let frameIconImage  = CGRect.init(origin: originIconImage, size: sizeIconImage)
+        
+        let iconImageView = UIImageView(image: UIImage(named: iconReferenceName, in: bundle, compatibleWith: nil))
+        
+        let identifierName = "\(iconReferenceName)SizeReferenceIdentifier"
+        setIdentifierForView(view: iconImageView, identifierName: identifierName)
+        iconImageView.frame                   = frameIconImage
+        
+        iconView.addSubview(iconImageView)
+        
+        return iconView
+    }
+    
+    private func createSeparatorView(yReference: CGFloat) -> UIView {
+        let separatorView   = UIView()
+        let originSeparator = CGPoint(x: 10, y: yReference)
+        let sizeSeparator   = CGSize(width: groupViewWidth - 20, height: 1)
+        let frameSeparator  = CGRect.init(origin: originSeparator, size: sizeSeparator)
+        
+        separatorView.frame           = frameSeparator
+        separatorView.backgroundColor = UIColor.black
+        separatorView.alpha           = 0.2
+        
+        return separatorView
+    }
+
+    private func configureGroupLayout(buttons: [Button], groupView: UIView, groupViewOirigin: CGPoint, addAsSubview: Bool = true) {
+        let _: [Button] = buttons.enumerated().map() { (index, button) in
+            let uiButton = button.uiButton
+            let size = uiButton.frame.size
+            
+            if index == 0 {
+                uiButton.frame = buttonFrame(y: 20, size: size)
+            } else {
+                let referencedButton = buttons[index - 1].uiButton
+                var yPlus: CGFloat = 75
+                if size != CGSize(width: 0, height: 0) {
+                    yPlus = 40
+                }
+                let y = referencedButton.frame.origin.y + yPlus
+                uiButton.frame = nextButtonFrame(referencedY: y, size: size)
+            }
+            
+            setImageEdgeInsets(button: uiButton)
+            setButtonImage(button: uiButton, imageName: button.imageName, imageColor: button.imageColor)
+            addButtonListener(uiButton, action: button.selector)
+            
+            groupView.addSubview(uiButton)
+            
+            return button
+        }
+        
+        let lastUiButton = buttons.last!.uiButton
+        
+        let height = lastUiButton.frame.origin.y + lastUiButton.frame.height + 20
+        let size   = CGSize(width: groupViewWidth, height: height)
+        let frame  = CGRect.init(origin: groupViewOirigin, size: size)
+        
+        groupView.frame              = frame
+        groupView.backgroundColor    = backgroundGroupGrayColor
+        
+        groupView.layer.cornerRadius = groupView.frame.width / 2
+        
+        if addAsSubview {
```

Comment:
I was taking a look here and you only use this function one time here, and the method is private. You are using this variable as false, so this conditional will never be rechearble.

Reply:
Removed. Thanks

Reply:
Chid, you removed the conditional.. But you are still adding the subview. My question is: you should do it?
Because on previous implementation, you had that conditional and you passed it as false.

Reply:
Sorry. Removed =D